lib/utils.ts

```diff
@@ -1,25 +1,128 @@
-import { Request } from 'express';
-
-export function rewriteLinkWithProxy(
-  originalUrl: string,
-  req: Request,
-): string {
-  const protocol = req.protocol;
-  const host = req.get('host');
-  const proxyBaseUrl = `${protocol}://${host}`;
+import { convertOparlToEli } from './convert';
+import { Parser, Store } from 'n3';
+import { enrichOparlDataToJsonLd } from './enrich';
+
+/**
+ * Changes the protocol and host of an URL with the protocol and host from the provided proxy URL. Also, adds the first segment of the path of the proxy URL.
+ * For example, https://ris.freiburg.de/oparl/System becomes http://localhost:8888/eli/oparl/System
+ * @param {string} originalUrl - URL that needs to be rewritten.
+ * @param {string} proxyUrl - URL of the proxy
+ * @returns {URL} URL object of the rewritten URL
+ */
+export function rewriteLinkWithProxy(originalUrl: string, proxyUrl: string): URL {
+  const originalUrlObj = new URL(originalUrl);
+  const proxyUrlObj = new URL(proxyUrl);
+
+  // replace host
+  const protocol = proxyUrlObj.protocol;
+  const host = proxyUrlObj.host;
+  let firstSegment = proxyUrlObj.toString().split('/')[1]; // 'eli' or 'oparl' or '' in harvesting
+  console.log('first segment: ' + firstSegment);
+  if (firstSegment != '') firstSegment = `/${firstSegment}`;
+
+  const newUrl = new URL(originalUrl.toString());
+  newUrl.protocol = protocol;
+  newUrl.host = host;
+  newUrl.pathname = `${firstSegment}${originalUrlObj.pathname}`;
   try {
-    const url = new URL(originalUrl);
-    return proxyBaseUrl + url.pathname + url.search;
+    console.log('new url: ' + newUrl.toString());
+    return newUrl;
   } catch (error) {
     console.error('Invalid URL:', originalUrl);
-    return originalUrl; // Fallback to the original URL if parsing fails
+    return originalUrlObj; // Fallback to the original URL if parsing fails
   }
 }
 
-export async function fetchJson(url) {
-  const response = await fetch(url);
-  console.log('Sent request:', url);
-  if (!response.ok) throw new Error(`OParl request failed: ${response.status}`);
+/**
+ * Removes the schema version from an Oparl schema URL string
+ * For example, https://schema.oparl.org/1.0/System becomes https://schema.oparl.org/System
+ * @param {string} url - Oparl schema URL 
+ * @returns {string} Oparl schema URL without version
+ */
+export function removeVersionFromOparlSchemaUri(url: string): string {
+  return url.replaceAll(/(schema\.oparl\.org)\/\d+\.\d+(?=\/|$)/g, '$1');
+}
 
+/**
+ * Fetch with logging and returning JSON
+ * @param {string} oparlUrl - Oparl API URL 
+ * @returns {object} JSON response
+ */
+export async function getOparlData(oparlUrl: string) {
```

Comment:
we're exporting all of these functions, but aren't a good part of these internal? best to keep them that way, it makes it more clear what the public API is of this file

Reply:
done https://github.com/lblod/oparl-to-eli-service/pull/4/commits/297db7e7c1e2061d41ca112debed06e0532d29aa